Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM node:18-alpine AS base
+ENV PNPM_HOME="/pnpm"
+ENV PATH="$PNPM_HOME:$PATH"
+RUN corepack enable
+
+FROM base AS build
+COPY . /usr/src/monorepo
+WORKDIR /usr/src/monorepo
+
+# required dependencies for node-gyp
+# context: https://github.com/nodejs/docker-node/blob/main/docs/BestPractices.md#node-gyp-alpine
+RUN apk add --no-cache python3 make g++
+
+# https://pnpm.io/cli/fetch
+# Fetches packages based on lockfile not package.json - cache is valid even if package.json changes
+RUN pnpm fetch
+
+# install monorepo dependencies, from the virtual store
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install --frozen-lockfile --offline
```

Comment:
`pnpm fetch` isn't helpful right now since we are only building one app. But this may become useful as we add more apps that may require copying over files in the build process that break the cache.

context: https://pnpm.io/cli/fetch